data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v2/validator/GetNewBlock.java

```diff
@@ -156,12 +156,10 @@ public void handleRequest(RestApiRequest request) throws JsonProcessingException
         provider.getUnsignedBeaconBlockAtSlot(slot, randao, graffiti, false);
     request.respondAsync(
         result.thenApply(
-            maybeBlock -> {
-              if (maybeBlock.isEmpty()) {
-                throw new ChainDataUnavailableException();
-              }
-              return AsyncApiResponse.respondOk(maybeBlock.get());
-            }));
+            maybeBlock ->
+                maybeBlock
+                    .map(AsyncApiResponse::respondOk)
+                    .orElseThrow(ChainDataUnavailableException::new)));
```

Comment:
not the problem of this ticket specifically, but im not sure this is correct, looks like generic handler will turn this to a 'no content' (204) and most of the time ChainDataUnavailable i would have thought should be a 503
the obvious answer might be to make this throw `ServiceUnavailableException` in the throw., or potentially the ChainDataUnavailable generic handler needs adjusting

Reply:
hmm, 503 is wrong as that means the node is syncing but that's not the case here. This is the same exception we through when we're pre-genesis and so have nothing to return.  Given this goes to combinedChainDataClient to get the required state and it's done by slot, I think the only case where this can come back empty is if we're pre-genesis which would make it consistent with other APIs (not sure it's *right* but it's consistent).